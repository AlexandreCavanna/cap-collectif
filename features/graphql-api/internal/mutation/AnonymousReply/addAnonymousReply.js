/* eslint-env jest */
import '../../../_setup';

const AddAnonymousReplyMutation = /* GraphQL*/ `
    mutation AddAnonymousReplyMutation($input: AddAnonymousReplyInput!) {
        addAnonymousReply(input: $input) {
            reply {
                __typename
                id
                participantEmail
            }
            questionnaire {
                id
            }
            token
            errorCode
        }
    }
`;

const input = {
  "questionnaireId": "UXVlc3Rpb25uYWlyZTpxdWVzdGlvbm5haXJlQW5vbnltb3Vz",
  "responses": [
    {
      "value": "abc",
      "question": "UXVlc3Rpb246MTQwMw=="
    }
  ],
  "participantEmail": null
};

const propertyMatchers = {
  addAnonymousReply: {
    reply: {
      id: expect.any(String)
    },
    token: expect.any(String)
  },
}

describe('mutations.addAnonymousReply', () => {
  it('should create an anonymous reply', async () => {
    const response = await graphql(
      AddAnonymousReplyMutation,
      {
        input,
      },
      'internal'
    );

    expect(response).toMatchSnapshot(propertyMatchers);
  });

  it('should create an anonymous reply with an email', async () => {
    const response = await graphql(
      AddAnonymousReplyMutation,
      {
        input: {
          ...input,
          participantEmail: 'abc@cap-collectif.com'
        }
      },
      'internal'
    );

    expect(response).toMatchSnapshot(propertyMatchers);
  });

  it('should create an anonymous reply with a complex email', async () => {
    const response = await graphql(
      AddAnonymousReplyMutation,
      {
        input: {
          ...input,
          participantEmail: 'Cécile.péon@laposte.net'
        }
      },
      'internal'
    );

    expect(response).toMatchSnapshot(propertyMatchers);
  });

  it('should not create an anonymous reply with an invalid email', async () => {
    const response = await graphql(
      AddAnonymousReplyMutation,
      {
        input: {
          ...input,
          participantEmail: 'i am an invalid email'
        }
      },
      'internal'
    );

    expect(response).toMatchSnapshot();
  });
});
