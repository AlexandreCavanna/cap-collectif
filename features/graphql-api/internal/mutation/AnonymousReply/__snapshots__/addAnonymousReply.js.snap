// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutations.addAnonymousReply should create an anonymous reply 1`] = `
Object {
  "addAnonymousReply": Object {
    "errorCode": null,
    "questionnaire": Object {
      "id": "UXVlc3Rpb25uYWlyZTpxdWVzdGlvbm5haXJlQW5vbnltb3Vz",
    },
    "reply": Object {
      "__typename": "AnonymousReply",
      "id": Any<String>,
      "participantEmail": null,
    },
    "token": Any<String>,
  },
}
`;

exports[`mutations.addAnonymousReply should create an anonymous reply with an email 1`] = `
Object {
  "addAnonymousReply": Object {
    "errorCode": null,
    "questionnaire": Object {
      "id": "UXVlc3Rpb25uYWlyZTpxdWVzdGlvbm5haXJlQW5vbnltb3Vz",
    },
    "reply": Object {
      "__typename": "AnonymousReply",
      "id": Any<String>,
      "participantEmail": "abc@cap-collectif.com",
    },
    "token": Any<String>,
  },
}
`;

exports[`mutations.addAnonymousReply should create an anonymous reply with a complex email 1`] = `
Object {
  "addAnonymousReply": Object {
    "errorCode": null,
    "questionnaire": Object {
      "id": "UXVlc3Rpb25uYWlyZTpxdWVzdGlvbm5haXJlQW5vbnltb3Vz",
    },
    "reply": Object {
      "__typename": "AnonymousReply",
      "id": Any<String>,
      "participantEmail": "Cécile.péon@laposte.net",
    },
    "token": Any<String>,
  },
}
`;

exports[`mutations.addAnonymousReply should not create an anonymous reply with an invalid email 1`] = `
Object {
  "addAnonymousReply": Object {
    "errorCode": "INVALID_FORM",
    "questionnaire": Object {
      "id": "UXVlc3Rpb25uYWlyZTpxdWVzdGlvbm5haXJlQW5vbnltb3Vz",
    },
    "reply": null,
    "token": null,
  },
}
`;
